### PR TITLE
JDK 11.0.9 / 8.0.272+ compatible FunctionsTest#printThrowable()

### PR DESCRIPTION
**Pull request by res0nance**
See #5032 , this also affects java 8.

Tested on AdoptOpenJDK 1.8.0_272 and Amazon Coretto 11.0.9

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
Thanks for the patch!

**Comment by res0nance** [comment]
I realized there was a small bug for older java 11 versions, retested again with AdoptOpenJDK 11.0.8 as well

**Review by MarkEWaite — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

File: core/src/test/java/hudson/FunctionsTest.java [hunk 1]
```diff
@@ -510,24 +512,45 @@ public void printStackTrace(PrintWriter s) {
...
-            "\t[CIRCULAR REFERENCE:p.Exc1]\n",
-            "<cycle to p.Exc1>\n" +
-            "Caused: p.Exc2\n" +
-            "\tat p.C.method2(C.java:27)\n" +
-            "Caused: p.Exc1\n" +
-            "\tat p.C.method1(C.java:17)\n");
-        */
+        //Format changed in 11.0.9 / 8.0.272 (JDK-8226809 / JDK-8252444 / JDK-8252489)
+        if ((getVersion().getDigitAt(0) == 11 && getVersion().isNewerThanOrEqualTo(new VersionNumber("11.0.9"))) ||
+                (getVersion().getDigitAt(0) == 8 && getVersion().isNewerThanOrEqualTo(new VersionNumber("8.0.272")))) {
```
Comment (jglick) [inline core/src/test/java/hudson/FunctionsTest.java:522]: Seems pretty fragile. Maybe better to just alter the `assertPrintThrowable` utility to tolerate either format.

**Pull request merged**